aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncClient.java

```diff
@@ -703,10 +798,74 @@ public boolean isMutationQueueEmpty() {
     }
 
     /**
-     * Clear the mutation queue. A Mutation that is currently in progress will continue to execute until finished.
+     * Clear the mutation queue. A Mutation that is currently in progress will
+     * continue to execute until finished.
      *
+     * @deprecated Please use
+     *     #clearCache(ClearCacheOptions.builder().clearMutations().build()) instead.
      */
+    @Deprecated
     public void clearMutationQueue() {
         mAppSyncOfflineMutationManager.clearMutationQueue();
     }
+
+    /**
+     * Clear the store created for Delta Sync. This method deletes all the records
+     * that stores the lastSyncTime of the sync queries.
+     */
+    private void clearDeltaSyncStore() {
+        Log.d(TAG, "Clearing the delta sync store.");
+
+        AWSAppSyncDeltaSyncSqlHelper awsAppSyncDeltaSyncSqlHelper =
+                new AWSAppSyncDeltaSyncSqlHelper(
+                    applicationContext,
+                    deltaSyncSqlStoreName);
+        new AWSAppSyncDeltaSyncDBOperations(awsAppSyncDeltaSyncSqlHelper)
+                .clearDeltaSyncStore();
+    }
+
+    /**
+     * Clears the different client databases on the local device
+     * which stores the following:
+     *
+     * 1) Query Cache - query responses
+     * 2) Mutation Queue - offline persistent mutations
+     * 3) Delta Sync Metadata Cache - Subscriptions Metadata
+     */
+    public void clearCache() {
+        clearCache(ClearCacheOptions.builder()
+                .clearQueries()
+                .clearMutations()
+                .clearSubscriptions()
+                .build());
+    }
+
+    /**
+     * Clears the different client databases on the local device
+     * based on the ClearCacheOptions object passed in.
+     *
+     * 1) ClearCacheOptions#clearQueries - Query Cache - query responses
+     * 2) ClearCacheOptions#clearMutations - Mutation Queue - offline persistent mutations
+     * 3) ClearCacheOptions#clearSubscriptions - Delta Sync Metadata Cache - Subscriptions Metadata
+     */
+    public void clearCache(ClearCacheOptions clearCacheOptions) {
+        try {
+            if (clearCacheOptions.isQueries()) {
+                Log.d(TAG, "Clearing the query cache.");
+                mSyncStore.clearAll().execute();
+            }
+
+            if (clearCacheOptions.isMutations()) {
+                Log.d(TAG, "Clearing the mutations queue.");
+                clearMutationQueue();
+            }
+
+            if (clearCacheOptions.isSubscriptions()) {
+                Log.d(TAG, "Clearing the delta sync subscriptions metadata cache.");
+                clearDeltaSyncStore();
+            }
+        } catch (Exception ex) {
+            throw new RuntimeException("Error in clearing the cache.", ex);
```

Comment:
This will crash the App. Given that this is a new API, use a checked exception.

Reply:
Added the checked exception and updated the description and tests.

Reply:
LGTM